nsot/models/resource.py

```diff
@@ -40,16 +40,24 @@ def set_query(self, query, site_id=None, unique=False):
             objects = objects.filter(site=site_id)
 
         try:
-            attributes = util.parse_set_query(query)
+            query_fields = util.parse_set_query(query)
         except (ValueError, TypeError) as err:
             raise exc.ValidationError({
                 'query': err.message
             })
 
         resource_name = self.model.__name__
 
+        # Grabbing the concrete fields for this Resource
+        concrete_field_names = []
+        # The below call can be replaced with .get_concrete_fields_with_model()
+        # once we upgrade to >= 2.1
+        concrete_fields = self.model._meta.concrete_fields
+        for field in concrete_fields:
+            concrete_field_names.append(field.name)
```

Comment:
Lines 51-57 is pretty much identical to lines 134-139 in `attributes.py`. Can you please factor this into a method on the `Resource` class, e.g. `Resource.get_concrete_field_names()` that returns a list of the field names?

Reply:
Done.